### PR TITLE
(Preliminary) Changelog for the 0.10.0 version

### PR DESCRIPTION
**Pull request by jrfnl**
:warning: **Needs careful review!!!!!**

I've done a walk through of the first three pages of the closed PRs which AFAICS covers the merged PRs since Feb 1 when the last release was tagged.

I have *not* gone through the individual commits in the 0.10.0 PR branch.

Changes/adjustments to sniffs new to this release have not been listed individually.

@WordPress-Coding-Standards/admins Everyone, please carefully check through the list for:
* missed contributions
* correct description of the changes
* anything else which should be in the changelog

Please feel free to push directly to this branch to update it with your additions/changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,10 +4,60 @@ All notable changes to this project will be documented in this file.
 
 This projects adheres to [Semantic Versioning](http://semver.org/) and [Keep a CHANGELOG](http://keepachangelog.com/).
 
-## [Unreleased]
```
Comment (GaryJones) [inline CHANGELOG.md]: Add a new section above:
```
[Unreleased]

_Nothing yet._
```

That then provides a link to compare the 0.10.0 against HEAD.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,10 +4,60 @@ All notable changes to this project will be documented in this file.
...
+## [0.9.0] - 2016-08-22
```
Comment (GaryJones) [inline CHANGELOG.md]: This should be `[0.10.0]`, not `[0.9.0]`.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,10 +4,60 @@ All notable changes to this project will be documented in this file.
...
+- `WordPress.WhiteSpace.ControlStructureSpacing` should no longer throw error notices during live code review.
+- Errors will be no longer be thrown for methods not complying with the naming conventions when the class extends a parent class or implements an interface - `WordPress.NamingConventions.ValidFunctionName` sniff.
+
+### Housekeeping
+- The `WordPress-Core` ruleset is now ordered according to the handbook
+- The WPCS code base itself now complies with the WordPress-Core, -Extra and -Docs coding standards.
+- Various other code quality and code consistency improvements under the hood.
+
 
 ## [0.9.0] - 2016-02-01
```
Comment (GaryJones) [inline CHANGELOG.md]: The compare link for 0.10.0 needs to be added at the bottom.

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,10 +4,60 @@ All notable changes to this project will be documented in this file.
...
+- Bug in the `WordPress.WhiteSpace.ControlStructureSpacing.BlankLineAfterEnd` sniff which was incorrectly being triggered on last method of class.
+- Function name sniffs based on the `AbstractFunctionRestrictions` parent class will now do a case-insensitive function name comparison.
+- Function name sniffs in the `WordPress.PHP.DiscouragedFunctions` sniff will now do a case-insensitive function name comparison.
+- Whitelist comments directly followed by a PHP closing tag were not being recognized.
+- Some PHP Magic constants were not recognized by the `WordPress.XSS.EscapeOutput` sniff.
+- An error message suggesting Camel caps rather than the intended snake case format in the `WordPress.NamingConventions.ValidFunctionName` sniff.
+- `WordPress.WhiteSpace.ControlStructureSpacing` should no longer throw error notices during live code review.
+- Errors will be no longer be thrown for methods not complying with the naming conventions when the class extends a parent class or implements an interface - `WordPress.NamingConventions.ValidFunctionName` sniff.
+
+### Housekeeping
```
Comment (GaryJones) [inline CHANGELOG.md]: I think these three could all go under the Changed section, instead of introducing a new section.

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,10 +4,60 @@ All notable changes to this project will be documented in this file.
...
+- Removed the sniffing for the `extract()` and POSIX functions from the `WordPress.VIP.RestrictedFunctions` sniff as these now have their own sniffs.
+
+### Fixed
+- Error message precision for the `WordPress.NamingConventions.ValidVariableName` sniff.
+- Bug in the `WordPress.WhiteSpace.ControlStructureSpacing.BlankLineAfterEnd` sniff which was incorrectly being triggered on last method of class.
+- Function name sniffs based on the `AbstractFunctionRestrictions` parent class will now do a case-insensitive function name comparison.
+- Function name sniffs in the `WordPress.PHP.DiscouragedFunctions` sniff will now do a case-insensitive function name comparison.
+- Whitelist comments directly followed by a PHP closing tag were not being recognized.
+- Some PHP Magic constants were not recognized by the `WordPress.XSS.EscapeOutput` sniff.
+- An error message suggesting Camel caps rather than the intended snake case format in the `WordPress.NamingConventions.ValidFunctionName` sniff.
```
Comment (GaryJones) [inline CHANGELOG.md]: Don't need a capital for `Camel caps` IMO.

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,10 +4,60 @@ All notable changes to this project will be documented in this file.
...
+- Reference to the [wiki](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki) to the Readme.
+- Recommendation to also use the [PHPCompatibility](https://github.com/wimg/PHPCompatibility) ruleset to the Readme.
+
+### Changed
+- The minimum required PHP_CodeSniffer version to 2.6.0.
+- Moved the `WordPress.WP.PreparedSQL` sniff from `WordPress-Extra` to `WordPress-Core`.
+- `WordPress.PHP.StrictInArray` will now also flag non-strict usage of `array_keys()` and `array_search()`.
+- Added `_deprecated_constructor()` and `_deprecated_hook()` to the list of printing functions.
+- Added numerous additional functions to sniff for to the `WordPress.VIP.RestrictedFunctions` sniff as per the VIP guidelines.
+- Upped the `posts_per_page` limit from 50 to 100 `WordPress.VIP.PostsPerPage` sniff as per the VIP guidelines.
```
Comment (GaryJones) [inline CHANGELOG.md]: Need "in", after "100".

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,10 +4,60 @@ All notable changes to this project will be documented in this file.
...
+- Added `__debugInfo` to the magic method whitelist for functions starting with double underscore in the `WordPress.NamingConventions.ValidFunctionName` sniff.
```
Comment (GaryJones) [inline CHANGELOG.md]: Considering the double underscore check was added in this release (as per line 34), this particular change seems redundant?

Comment (jrfnl) [inline CHANGELOG.md]: Will clarify the lines. The `debugInfo` bit is about double underscore **_methods_**, which is a check which was already being done. Line 34 is about double underscored prefixed **_functions_**, i.e. global namespace.

**Comment by jrfnl** [comment]
Updated to address all @GaryJones's comments.

Oh and I forgot to mention above that I set a preliminary release date for this Monday. If the release will be done before/after that date, let's make sure the date will be adjusted to match.

File: CHANGELOG.md [hunk 8]
```diff
@@ -6,8 +6,60 @@ This projects adheres to [Semantic Versioning](http://semver.org/) and [Keep a C
...
+## [0.10.0] - 2016-08-22
+
 ### Added
-- Sniff to flag dynamic translatable strings and textdomains.
+- `WordPress.WP.I18n` sniff to the `WordPress-Core` ruleset to flag dynamic translatable strings and textdomains.
+- `WordPress.PHP.DisallowAlternativePHPTags` sniff to the `WordPress-Core` ruleset to flag - and fix - ASP and `<script>` PHP open tags.
+- `WordPress.Classes.ClassOpeningStatement` sniff to the `WordPress-Core` ruleset to flag - and fix - class opening brace placement.
+- `WordPress.NamingConventions.ValidHookName` sniff to the `WordPress-Core` ruleset to flag filter and action hooks which don't comply with the guideline of lowercase letters and underscores. For maintaining backward-compatibility of hook names an `additionalWordDelimiters` property can be added via a custom ruleset.
+- `WordPress.Functions.DontExtract` sniff to the `WordPress-Core` ruleset to flag usage of the `extract()` function.
+- `WordPress.PHP.POSIXFunctions` sniff to the `WordPress-Core` ruleset to flag usage of regex functions from the POSIX PHP extension which was deprecated since PHP 5.3 and removed in PHP 7.
```
Comment (grappler) [inline CHANGELOG.md]: Did we want to mention that they were removed from `WordPress.PHP.DiscouragedFunctions`?

Comment (jrfnl) [inline CHANGELOG.md]: I already do in the `Removed` section.

File: CHANGELOG.md [hunk 9]
```diff
@@ -6,8 +6,60 @@ This projects adheres to [Semantic Versioning](http://semver.org/) and [Keep a C
...
+- Updated installation instructions in the readme.
+- The `WordPress-Core` ruleset is now ordered according to the handbook
+- The WPCS code base itself now complies with the WordPress-Core, -Extra and -Docs coding standards.
+- Various other code quality and code consistency improvements under the hood.
+
+### Removed
+- `Squiz.Functions.FunctionDeclarationArgumentSpacing.SpacingBeforeClose` from the `WordPress-Core` standard (was causing duplicate messages for the same issue).
+- `Squiz.Commenting.FunctionComment.ScalarTypeHintMissing`, `Squiz.Commenting.InlineComment.NotCapital` from the `WordPress-Docs` standard.
+- Removed the sniffing for `get_pages()` from the `WordPress.VIP.RestrictedFunctions` sniff as per the VIP guidelines.
+- Removed the sniffing for the `extract()` and POSIX functions from the `WordPress.VIP.RestrictedFunctions` sniff as these now have their own sniffs.
```
Comment (grappler) [inline CHANGELOG.md]: We should mention that we removed them from `WordPress.PHP.DiscouragedFunctions` too.

Comment (jrfnl) [inline CHANGELOG.md]: Fixed.

**Comment by jrfnl** [comment]
So... the release date of yesterday is not going to happen. What shall I change it to ? Or maybe I should ask: who has time to look at the 0.10.0 branch and this PR and review it ?

**Comment by westonruter** [comment]
@jrfnl I've bumped the release date to this Monday instead. Sorry for the delay. You have done _amazing_ work. 👏 You've demonstrate a great command over the PHP_CodeSniffer and this repo. The last thing I want is for you to be blocked and lose your momentum. You have my vote of confidence. I have zero reservations about you merging PRs and doing releases yourself, so you now can merge into `master` yourself. If you want to handle the release of 0.10.0 yourself, more power to you. Otherwise, I should be able to do the release on Monday.

**Comment by jrfnl** [comment]
@westonruter Thank you for your kind words & confidence. Will try to use my new powers for good :sunglasses:

**Pull request merged**